src/clock.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * FuzzJudge - Randomised input judging server, designed for ProgComp.
+ * Copyright (C) 2024 UTS Programmers' Society (ProgSoc)
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+interface TimeConfig {
+  start: Date;
+  freeze: Date;
```

Comment:
imho freeze should probably be "halt", just to avoid crossing terminologies with the scoreboard freeze

Reply:
I had intended that that was the time the scoreboard freeze happens, what do you mean by crossing terminologies?

Reply:
oh i see yeah that's fair, would be good to have a halt time if we need to pause the competition for some reason.